Sources/AuthFoundation/User Management/Credential.swift

```diff
@@ -157,13 +186,20 @@ public final class Credential: Equatable, OAuth2ClientDelegate {
     /// Indicates this credential's token should automatically be refreshed prior to its expiration.
     ///
     /// This property can be used to ensure a token is available for use, by refreshing the token automatically prior to its expiration. This uses the ``Credential/refreshGraceInterval`` in conjunction with the current ``TimeCoordinator`` instance, to refresh the token before its scheduled expiration.
-    public var automaticRefresh: Bool = false {
-        didSet {
-            guard oldValue != automaticRefresh else { return }
-            if automaticRefresh {
-                startAutomaticRefresh()
-            } else {
-                stopAutomaticRefresh()
+    public var automaticRefresh: Bool {
+        get {
+            lock.withLock { _automaticRefresh }
+        }
+        set {
+            lock.withLock {
+                guard _automaticRefresh != newValue else { return }
+                _automaticRefresh = newValue
+
+                if _automaticRefresh {
+                    startAutomaticRefresh()
```

Comment:
How long this can execute? Note, the block still acquires lock. Is it that necessary to keep property locked until function is not finished?

Reply:
This function is improved upon within a subsequent PR, but ultimately starting and stopping the timer is a very short operation, since the body of the function almost immediately dispatches to a cancellable task.